src/plugins/postgres_check_operator.py

```diff
@@ -1,139 +1,305 @@
+     ORDER BY column_name
+""",
+    result_fetcher=flattened_records_as_set,
+)
 
-    :param sql: the sql to be executed
-    :type sql: str
-    :param postgres_conn_id: reference to the Postgres database
-    :type postgres_conn_id: str
+GEO_CHECK = CheckFactory(
+    """
+  {% set lparams = params['$check_id'] %}
+  {% set geo_column = lparams.geo_column|default("geometry", true) %}
+  SELECT 1 WHERE NOT EXISTS (
+      SELECT FROM {{ lparams.table_name }} WHERE
+        {{ geo_column }} IS null
+        {% if lparams.check_valid|default(true) %} OR ST_IsValid({{ geo_column }}) = false {% endif %}
+          OR GeometryType({{ geo_column }})
+        {% if lparams.geotype is string %}
+          <> '{{ lparams.geotype }}'
+        {% else %}
+          NOT IN ({{ lparams.geotype | map('quote') | join(", ") }})
+        {% endif %}
+      )
+""",
+    result_fetcher=first_flattened_record,
+)
+
+
+class PostgresMultiCheckOperator(BaseOperator):
+    """ This operator can be used to fire a number of checks at once.
+        This is more efficient than firing up all checks separately.
+        There is one caveat. Because we want to use the efficient and
+        lazily evaluated jinja templating, the template parameters have
+        to be collected into one dict, because Airflow does the parameter
+        interpolation only once. So, when params are used, they have to
+        be collected with the 'make_params' function and fed into the
+        operator constructor.
     """
 
-    template_fields = ("sql",)
-    template_ext = (".sql",)
+    # We use the possibilty to have nested template fields here
+    template_fields: Iterable[str] = ["checks"]
 
     @apply_defaults
-    def __init__(self, sql, postgres_conn_id="postgres_default", *args, **kwargs):
-        super(PostgresCheckOperator, self).__init__(sql=sql, *args, **kwargs)
+    def __init__(
+        self, postgres_conn_id="postgres_default", checks=[], *args, **kwargs,
+    ):
+        super(PostgresMultiCheckOperator, self).__init__(*args, **kwargs)
         self.postgres_conn_id = postgres_conn_id
-        self.sql = sql
+        self.checks = checks
+        self.check_checks_for_params(kwargs.get("params"))
+
+    def check_checks_for_params(self, params):
+        if not params and reduce(
+            lambda acc, d: {**acc, **(d or {})},
+            list(make_params(self.checks).values()),
+            {},
+        ):
+            raise ValueError("One of the checks needs params")
+
+    def execute(self, context=None):
+
+        hook = self.get_db_hook()
+
+        for checker in self.checks:
+            self.log.info(
+                "Executing SQL check: %s with %s", checker.sql, repr(checker.parameters)
+            )
+            records = hook.get_records(checker.sql, checker.parameters)
+
+            if not records:
+                raise AirflowException("The query returned None")
+
+            checker_fie = checker.result_checker or operator.eq
```

Comment:
wat is `fie`?

Reply:
:-) Soort generieke naam voor een functie. Maar komt meer uit de wiskunde. Niet zo duidelijk, ik zal er function van maken.